src/intel_pmu.c

```diff
@@ -285,22 +285,20 @@ static void pmu_submit_counter(int cpu, char *event, counter_t value,
   plugin_dispatch_values(&vl);
 }
 
-meta_data_t *pmu_event_get_meta(struct event *e, int cpu) {
+meta_data_t *pmu_meta_data_create(const struct efd *efd) {
   meta_data_t *meta = NULL;
 
   /* create meta data only if value was scaled */
-  if (e->efd[cpu].val[1] != e->efd[cpu].val[2] && e->efd[cpu].val[2]) {
+  if (efd->val[1] != efd->val[2] && efd->val[2]) {
```

Comment:
Please invert the logic to remove one level of indentation:

```c
if (efd->val[1] == efd->val[2] || !efd->val[2]) {
  return NULL;
}

meta_data_t *meta = meta_data_create();
/* … */
```

Reply:
Ok